nameko_eventlog_dispatcher/eventlog_dispatcher.py

```diff
@@ -52,32 +52,33 @@ def worker_setup(self, worker_ctx):
             return
 
         try:
-            self._get_dispatch(worker_ctx)(event_type=self.ENTRYPOINT_FIRED)
+            dispatcher = self.get_dependency(worker_ctx)
+            dispatcher(event_type=self.ENTRYPOINT_FIRED)
         except Exception as exc:
             log.error(exc)
 
     def get_dependency(self, worker_ctx):
-        return self._get_dispatch(worker_ctx)
-
-    def _get_dispatch(self, worker_ctx):
-        headers = self.get_message_headers(worker_ctx)
-
-        def dispatch(event_type, event_data=None, metadata=None):
-            body = self._get_base_call_info(worker_ctx)
-            body.update(metadata or {})
-            body['timestamp'] = _get_formatted_utcnow()
-            body['event_type'] = event_type
-            body['data'] = event_data or {}
-            self.publisher.publish(
-                body,
-                exchange=self.exchange,
-                routing_key=self.event_type,
-                extra_headers=headers
+        event_dispatcher = super().get_dependency(worker_ctx)
```

Comment:
Really nice move to get a solution relying on the Nameko implementation, abstracting even more internal pieces!